### ISSUE TITLE
Changing dtypes for size reduction

### ISSUE THREAD

**Issue by mgeraeds** [mgeraeds/hires-processing#12]
A useful data reduction method would be to change dtypes, e.g. from `float64` to `float32` in big datasets. This alone can already allow for a size reduction of 50%. Two things can be tested to find out the ideal configuration for this:

- [x] Adding keyword argument for dtype change in reading/writing function (and manually putting in dtypes in function);
- [x] Providing an `encoding_dict` in the call to `xr.UgridDataset.ugrid.to_netcdf`.

**Comment by mgeraeds** [comment]
Using `encoding_dict` seems to be a good option when changing the `dtype`. Further testing steps include:

- [ ] Check how to implement a `None` encoding parameter or otherwise an empty dictionary (can the `ugrid.to_netcdf()` function handle this)

For wider applicability, it is chosen that a reduction function should apply to both native `xr.Datasets` as well as `xu.core.UgridDatasets`. When working with unstructured grids and the `xugrid` package, some additional "hidden" variables need to be written as well, such as the gridname and the connectivity variables (`mesh2d`, `mesh2d_face_nodes` and `mesh2d_edge_nodes` in recent versions of D-FLOW FM).

Next steps include:
- [ ] Write up a function with command line arguments for easy access (new issue) that takes both partitioned or unpartitioned data and both unstructured and structured datasets (but no combination).

**Comment by mgeraeds** [comment]
The reading/writing of reduced data turned out to be slow. What we ran into:

1. The data can't be appended to an earlier file in parallel (due to restricted  read/write access of HDF5 and NetCDF4 libraries);
2. Writing per variable does not contain all necessary topology for a fully functioning unstructured dataset - when writing a variable that only exists on the edges, for example, the written dataset will not contain the face locations.

Therefore, to circumvent these issues, we've implemented the `dytpe` change in parallel, writing the reduced data per timestep instead of per variable. There are downsides to this, mainly related to the size of the "overhead data" (topology-related variables that area repeated) but since this works for now, I opened a new issue (#16) for per-variable writing as an enhancement.
